salt/states/user.py

```diff
@@ -676,6 +676,14 @@ def present(name,
                                          ' {1}'.format(name, 'XXX-REDACTED-XXX')
                         ret['result'] = False
                     ret['changes']['password'] = 'XXX-REDACTED-XXX'
+                if empty_password:
+                    __salt__['shadow.del_password'](name)
```

Comment:
Not really.

If we are in the case of creating a user, L457 will try to remove the password of a non-existent user. If the user already exists, the password will be removed, without checking that it is necessary.

I modified the commit to verify that it is necessary and handle both cases:
 * Creating a user
 * User already present

There is now the password removing information in the changes.